src/CMakeLists.txt

```diff
@@ -462,6 +466,10 @@ ADD_SUBDIRECTORY ( Commands )
 ADD_SUBDIRECTORY ( Desktop )
 ADD_SUBDIRECTORY ( CommandLine )
 ADD_SUBDIRECTORY ( Tests )
+if (CMAKE_COMPILER_IS_GNUCC)
```

Comment:
Yes, in order to use the AVX intrinsics you have to compile with -mavx. If you compile with -mavx, the compiler can also generate AVX instructions elsewhere (not only in those places where you explicitly use the intrinsics). This is why I had to use the dispatcher and compile dot_avx etc separately in the first place. In that regard, It's also a pity that there is no flag to tell the compiler to use the AVX instructions only in those places where you use the intrinsics but nowhere else.